stake-pool/cli/src/main.rs

```diff
@@ -421,50 +757,25 @@ fn pick_withdraw_accounts(
     let mut withdraw_from: Vec<WithdrawAccount> = vec![];
     let mut remaining_amount = amount;
 
-    let mut split_candidate: Option<WithdrawAccount> = None;
-
-    // If amount to withdraw is more than the largest account
-    // then use up all the large accounts first
-    // Then either find the smallest account with balance more than remaining
-    // or find account to claim with exactly the same balance as remaining
+    // Go through available accounts and withdraw from largest to smallest
     for (pubkey, account) in accounts {
-        if account.lamports <= remaining_amount {
-            if split_candidate.is_some() {
-                // If there is an active split candidate and current account is smaller than remaining
-                // then break to use split candidate
-                if account.lamports < remaining_amount {
-                    break;
-                }
-                // Otherwise (account balance == remaining) procede with `claim`
-            }
-            // Those accounts will be withdrawn completely with `claim` instruction
-            withdraw_from.push(WithdrawAccount {
-                pubkey: *pubkey,
-                account: account.clone(),
-                amount: account.lamports,
-            });
-            remaining_amount -= account.lamports;
-            if remaining_amount == 0 {
-                // We filled all the balance, ignore split candidate
-                split_candidate = None;
-                break;
-            }
-        } else {
-            // Save the last account with balance more than remaining as a candidate for split
-            split_candidate = Some(WithdrawAccount {
-                pubkey: *pubkey,
-                account: account.clone(),
-                amount: remaining_amount,
-            });
+        if account.lamports <= *MIN_STAKE_BALANCE {
+            continue;
         }
-    }
-
-    // If there is a pending account to split, add it to the list
-    if let Some(withdraw) = split_candidate {
-        remaining_amount -= withdraw.amount;
+        let available_for_withdrawal = account.lamports - *MIN_STAKE_BALANCE;
```

Comment:
This needs to take rent exemption into account as well (should be fixed with `MIN_STAKE_BALANCE` though)

Reply:
See my comments above.